manifests/interface.pp

```diff
@@ -79,45 +79,52 @@
 
     # Setup interfaces for the slaves
     $bonding_slaves.each |String $ifname| {
-      network::interface { $ifname:
+      network_config { $ifname:
         ensure               => $ensure,
-        enable               => $enable,
         onboot               => $onboot,
-        type                 => 'InfiniBand',
         master               => $name,
         slave                => 'yes',
-        nm_controlled        => $_nm_controlled,
         mtu                  => $mtu,
-        options_extra_redhat => $options_extra_redhat,
```

Comment:
`$options_extra_redhat` needs to be kept but could likely just be moved directly into the `options` hash by setting `CONNECTED_MODE => $connected_mode`.

Reply:
Okay, I've added those to the options hash.